packages/react-redux/src/slices/cartoSlice.js

```diff
@@ -280,8 +280,13 @@ export const setViewState = (viewState) => {
       delete viewState[viewProp];
     }
 
-    dispatch(_setViewState(viewState));
-    debouncedSetViewPort(dispatch, _setViewPort);
+    const { carto: currentState } = getState();
+    const newViewState = { ...currentState.viewState, ...viewState };
+    // Deep compare current viewState and the new one.
+    if (!dequal(currentState.viewState, newViewState)) {
```

Comment:
I no totally agree with this comparison. setViewState only be called if the `viewState` has changed so if the client that uses c4r is calling this function with the same value the problem is on the client-side and we should fix it in the template

Reply:
When you are moving a layer in nebula.gl, deck.gl calls onViewStateChange with the same viewState. This can also happen in some other cases.

Reply:
I'm a bit worried that we are masking the error. We're going to research about this